clicommand/meta_data_get.go

```diff
@@ -59,6 +60,12 @@ var MetaDataGetCommand = cli.Command{
 			Usage:  "Which job's build should the meta-data be retrieved from",
 			EnvVar: "BUILDKITE_JOB_ID",
 		},
+		cli.StringFlag{
+			Name:   "build",
+			Value:  "",
+			Usage:  "Which build should the meta-data be retrieved from. --build will take precedence over --job",
+			EnvVar: "BUILDKITE_METADATA_BUILD_ID",
```

Comment:
just to write down a conversation we had over zoom yesterday:

originally, the value of this envar was `BUILDKITE_BUILD_ID`, but it being this would cause all metadata queries to be made at the build level, as the `BUILDKITE_BUILD_ID` envar is set on every build. Changing the envar here means that the user has to specifically say that they'd like build metadata.